platform/core.windows/src/org/netbeans/core/windows/options/LafPanel.java

```diff
@@ -296,7 +297,11 @@ private void switchEditorColorsProfile() {
         try {
             Class klz = cl.loadClass( COLOR_MODEL_CLASS_NAME );
             Object colorModel = klz.newInstance();
-            Method m = klz.getDeclaredMethod( "setCurrentProfile", String.class ); //NOI18N
+            Method m = klz.getDeclaredMethod( "getAnnotations", String.class ); //NOI18N
```

Comment:
Wow! That is ugly. Isn't the UI misplaced when it needs to use reflection to load a class from other module? If not, then ideally there would be an SPI interface `CurrentProfile` and one would use `Lookup.getDefault().lookupAll(CurrentProfile.class).forEach(it -> it.setProfile(preferredProfile))`.

Reply:
That's true, filed [NETBEANS-4039](https://issues.apache.org/jira/browse/NETBEANS-4039) for this. And yes, we blindly followed the pattern was used before us.